Remotion/Web/Development.WebTesting/BrowserSession/Chrome/ChromeBrowserSession.cs

```diff
@@ -1,140 +1,64 @@
-          Thread.Sleep ((int) sleep);
-          sleep *= 1.25f;
-
-          tries++;
-        }
-      } while (tries < maxTries);
-    }
-
-    /// <summary>
-    /// Removes the user directory root if we are the last session to use it. 
-    /// </summary>
-    private void DeleteUserDirectoryRootOnDemand ()
-    {
-      var userDirectoryRoot = BrowserConfiguration.UserDirectoryRoot;
-
-      if (!BrowserConfiguration.EnableUserDirectoryRootCleanup)
-        return;
-
-      if (string.IsNullOrEmpty (userDirectoryRoot))
-        return;
-
-      if (!Directory.Exists (userDirectoryRoot))
-        return;
-
-      if (Directory.GetDirectories (userDirectoryRoot).Length > 0)
-        return;
-
-      Directory.Delete (userDirectoryRoot);
-    }
-  }
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using OpenQA.Selenium;
+using Remotion.Web.Development.WebTesting.WebDriver.Configuration;
+using Remotion.Web.Development.WebTesting.WebDriver.Configuration.Chrome;
+
+namespace Remotion.Web.Development.WebTesting.BrowserSession.Chrome
+{
+  /// <summary>
+  /// Implements <see cref="IBrowserSession"/> for the Chrome browser.
+  /// </summary>
+  public class ChromeBrowserSession : BrowserSessionBase<IChromeConfiguration>
+  {
+    private readonly IBrowserSessionCleanUpStrategy[] _cleanUpStrategies;
+
+    public ChromeBrowserSession (
+        [NotNull] Coypu.BrowserSession value,
+        [NotNull] IChromeConfiguration configuration,
+        int driverProcessID,
+        [CanBeNull] [ItemNotNull] IBrowserSessionCleanUpStrategy[] cleanUpStrategies = null)
+        : base (value, configuration, driverProcessID)
+    {
+      _cleanUpStrategies = cleanUpStrategies;
+    }
+
+    /// <inheritdoc />
+    public override IReadOnlyCollection<BrowserLogEntry> GetBrowserLogs ()
+    {
+      return ((IWebDriver) Driver.Native).Manage().Logs.GetLog (LogType.Browser)
+          .Select (logEntry => new BrowserLogEntry (logEntry))
+          .ToArray();
+    }
+
+    /// <inheritdoc />
+    public override void Dispose ()
+    {
+      base.Dispose();
+
+      if (_cleanUpStrategies != null)

```

Comment:
Can this be null?
(we don't have a finlizer on this type, so...?)

Reply:
`_cleanUpStrategies` is an optional parameter, allowing it to be `null`. Shouldn't this be the case here?

Reply:
It is? oh, yeah. My bad.

Reply:
Hmm...why can a collection be nullable? Probably why I didn't realize that it was optional.

collection should be empty but not null after the ctor has completed. Use a static empty array if you want to do memory optimization.